articles/image-classifier-keras/index.md

```diff
@@ -0,0 +1,488 @@
+
+Image Augmentation helps us increase the size of our training set. It will also help to reduce overfitting. *The ImageDataGenerator()* class generates batches of tensor image data with real-time data augmentation. We will create objects of the class. 
+
+One will be for the training images, where we will apply image augmentation. We will also create objects for the validation images and the test images.
+
+We begin by rescaling the images, which normalizes the pixel values of our images. The *rescale* parameter is for feature scaling. It is vital when training neural networks. The *validation_split* parameter allows us to split a subset of our training data into the validation set. 0.2 means we will use 20% of our training set as the validation set. We will set *zoom_range* to 0.2. Finally, we set *horizontal_flip* to True.
+
+Now that we have created the objects, we need to connect them to our dataset.  The ImageDataGenerator() class has a function called flow_from_directory. The function will connect the image augmentation tool to the images in our dataset. It takes the path to the directory and generate batches of augmented data.
+
+```python
+# Awesome HyperParameters!!!
+TARGET_SIZE = (200, 200)
+BATCH_SIZE = 32
+CLASS_MODE = 'binary'  # for two classes; categorical for over 2 classes
+
+# Connecting the ImageDataGenerator objects to our dataset
+train_generator = dgen_train.flow_from_directory(train_dir,
+                                                 target_size=TARGET_SIZE,
+                                                 subset='training',
+                                                 batch_size=BATCH_SIZE,
+                                                 class_mode=CLASS_MODE)
+
+validation_generator = dgen_train.flow_from_directory(train_dir,
+                                                      target_size=TARGET_SIZE,
+                                                      subset='validation',
+                                                      batch_size=BATCH_SIZE,
+                                                      class_mode=CLASS_MODE)
+test_generator = dgen_test.flow_from_directory(test_dir,
+                                               target_size=TARGET_SIZE,
+                                               batch_size=BATCH_SIZE,
+                                               class_mode=CLASS_MODE)
+```
+
+![alt text](/engineering-education/image-classifier-keras/11.jpg)
+
+The first argument is the path to the dataset. The next parameter is the *target_size*. It will resize all the images to the specified target size of 200x200. The *batch size* defines how many images we want to have in each batch. 
+
+We will use a batch size of 32, and the class mode is either *binary* or *categorical*. *Binary* is for two output classes, while categorical is for more than two classes. Since we are working with only *two* classes, we will set the class mode to *binary*.

```

Comment:
Don't use future tense. Instead, write it as: We use a batch size of 32... I observed it throughout your article. Needs to be changed at all occurrences.